### PR TITLE
Updated node_pools variable documentation

### PR DESCRIPTION
**Pull request by neelesh9795**
Added the wrong file last time. Updated the autogen/README.md

### CODE REVIEW

File: autogen/README.md [hunk 1]
```diff
@@ -183,6 +183,24 @@ In order to operate with the Service Account you must activate the following API
 - Compute Engine API - compute.googleapis.com
 - Kubernetes Engine API - container.googleapis.com
 
+## node_pools Variable
+The node_pools variable takes the following parameters:
+
+| Name | Description | Requirement | 
+| --- | --- | --- |
+| name | The name of the node pool. If left blank, Terraform will auto-generate a unique name | Optional |
+| min_count | Minimum number of nodes in the NodePool. Must be >=0 and <= max_count | Required |
```
Comment (morgante) [inline autogen/README.md]: Not required, has a default: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/cluster.tf#L150

File: autogen/README.md [hunk 2]
```diff
@@ -183,6 +183,24 @@ In order to operate with the Service Account you must activate the following API
...
+| max_count | Maximum number of nodes in the NodePool. Must be >= min_count | Required |
```
Comment (morgante) [inline autogen/README.md]: Not required.

**Comment by morgante** [comment]
@neelesh9795 https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/autogen/cluster.tf#L299

**Comment by aaron-lane** [comment]
This should incorporate the addition of `node_locations` in #303.

**Comment by thecodejunkie** [comment]
Does `auto_upgrade` actuall have a default value of `false`?

I just tried this module with the following node pool definition

```
{
    autoscaling        = false
    name               = var.default_node_pool.name
    machine_type       = var.default_node_pool.machine_type
}
```

after I ran `terraform plan` I could see the following in the output for the node pool


```
+management {
  + auto_repair  = true
  + auto_upgrade = true
}
```

but if I explicitly add `auto_upgrade = false` then I correctly displayed as

```
+ management {
   + auto_repair  = true
   + auto_upgrade = false
}
```

**Comment by morgante** [comment]
The default depends on whether the cluster is regional or zonal: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/autogen/main.tf#L62

**Comment by thecodejunkie** [comment]
@morgante ah! Then I humble request that default values, that depend on whether the cluster is regional or zonal, are also reflected in this part of the documentation. Keep up the awesome work!

**Comment by morgante** [comment]
Agreed. @neelesh9795 please make sure to update with that.

**Comment by thecodejunkie** [comment]
@neelesh9795 please also consider sorting the table in alphabetical order 🤗

File: autogen/README.md [hunk 3]
```diff
@@ -182,6 +182,26 @@ In order to operate with the Service Account you must activate the following API
 - Compute Engine API - compute.googleapis.com
 - Kubernetes Engine API - container.googleapis.com
 
+## node_pools variable
```
Comment (aaron-lane) [inline autogen/README.md]: This table must include `node_count` which was added in #313.

**Comment by neelesh9795** [comment]
Closing this due to merge conflicts. Re opening at #331

**Pull request closed without merging**